SQL_Assignment.txt

```diff
@@ -0,0 +1,1093 @@
+| Grass    |
+| Cisneros |
++----------+
+
+49) Write two queries that will produce all orders taken on October 3 or October 4.
+
+mysql> select ONUM from Orders where Odate='90-3-10' or odate='90-4-10';
++------+
+| ONUM |
++------+
+| 3001 |
+| 3002 |
+| 3003 |
+| 3005 |
+| 3006 |
+| 3007 |
+| 3009 |
++------+
+
+50) Find only those customers whose ratings are higher than every customer in Rome.
+
+mysql> select CNAME from Customers where RATING >(select max(RATING) from Customers where CITY='Rome');
++----------+
+| CNAME    |
++----------+
+| Grass    |
+| Cisneros |
++----------+
+
+51) Write a query on the Customers table whose output will exclude all customers with a rating<= 100.00, unless they are located in Rome.
+
+52) Find all rows from the customer’s table for which the salesperson number is 1001.
+
+mysql> select * from Customers where SNUM=1001;
++------+---------+--------+--------+------+
+| CNUM | CNAME   | CITY   | RATING | SNUM |
++------+---------+--------+--------+------+
+| 2001 | Hoffman | Londan |    100 | 1001 |
+| 2006 | Clemens | London |    100 | 1001 |
++------+---------+--------+--------+------+
+
+53) Find the total amount in orders for each salesperson where their total of amounts are greater than the amount of the largest order in the table.
+
+54) Write a query that selects all orders save those with zeroes or NULL in the amount file.
+
+select ONUM from Orders where AMT is NULL or AMT=0;
+Empty set (0.00 sec)
+
+55) Produce all combinations of salespeople and customer names such that the former precedes the latter alphabetically, and the latter has a
+rating of less than 200.
+
+mysql> select s.SNAME,c.CNAME,c.RATING from Customers c,Salespeople s where c.RATING<200 group by s.SNAME,c.CNAME,c.RATING having c.CNAME>=s.SNAME;
++---------+---------+--------+
+| SNAME   | CNAME   | RATING |
++---------+---------+--------+
+| AxelRod | Clemens |    100 |
+| AxelRod | Hoffman |    100 |
+| AxelRod | Pereira |    100 |
+| Fran    | Hoffman |    100 |
+| Fran    | Pereira |    100 |
+| Motika  | Pereira |    100 |
+| Peel    | Pereira |    100 |
++---------+---------+--------+
+56) Find all salespeople name and commission.
+
+mysql> select Salespeople.SNAME,sum(COMM*AMT*0.01) from Salespeople,Orders where Salespeople.SNUM=Orders.SNUM group by SNAME;
++---------+--------------------+
+| SNAME   | sum(COMM*AMT*0.01) |
++---------+--------------------+
+| AxelRod |             171.32 |
+| Motika  |             209.01 |
+| Peel    |            1845.85 |
+| Rifkin  |             167.53 |
+| Serres  |             851.00 |
++---------+--------------------+
+
+57) Write a query that produces the names and cities of all customers with the same rating as Hoffman. Write the query using Hoffman’s cnum
+rather than his rating, so that it would still be usable if his rating is changed.
+
+mysql> select CNAME,CITY from Customers where RATING in(select RATING from Customers where SNUM = (select SNUM from Customers where CNAME='Hoffm

```

Comment:
This is a wrong query. We should use Hoffman's "cnum" and then try to produce the results accordingly.